File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> m;

        for(int i=0; i < nums.size(); i++){
            if(m.find(nums[i]) == m.end()){
                m[target - nums[i]] = i;
            }
            else{
                return {m[nums[i]], i};
            }
        }

        // for(auto it: m){
        //     cout<<it.first<<" "<<it.second<<endl;
        // }
        // return {0, 1};
    }
};

File: Data Structures & Algorithms/search-2d-matrix/submission-0.py
class Solution:
    def bS(self, nums, target):
        
        l = 0
        r = len(nums) - 1
        while(l <= r):
            mid = (l + r) // 2
            if nums[mid] < target:
                l = mid + 1
            elif nums[mid] > target:
                r = mid - 1
            else:
                return True
        
        return False

    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        
        for nums in matrix:
            if self.bS(nums, target):
                return True
        
        return False